numpy/matrixlib/defmatrix.py

```diff
@@ -422,7 +422,7 @@ def tolist(self):
         return self.__array__().tolist()
 
     # To preserve orientation of result...
-    def sum(self, axis=None, dtype=None, out=None):
+    def sum(self, axis=None, dtype=None, out=None, keepdims=True):
```

Comment:
Hmm, `keepdims` has no effect, so I suppose that this is here for compatibility with ndarray. I think it is better to not have the argument simply to avoid confusion. Perhaps this is worth a post on the list for discussion.